pandas/tseries/offsets.py

```diff
@@ -689,11 +754,28 @@ def rollforward(self, dt):
                 return self._prev_opening_time(dt)
         return dt
 
+    def _get_closing_time(self, dt):
+        """
+        Get the closing time of a business hour interval by its opening time.
+
+        Parameters
+        ----------
+        dt : datetime
+            Opening time of a business hour interval.
+
+        Returns
+        -------
+        result : datetime
+            Corresponding closing time.
+        """
+        for i, st in enumerate(self.start):
+            if st.hour == dt.hour and st.minute == dt.minute:
+                return dt + timedelta(
+                    seconds=self._get_business_hours_by_sec(st, self.end[i]))
+        assert False
```

Comment:
Presumably this shouldn't be reachable?  ValueError?

Reply:
Yes, this is not reachable. Previously I raise ValueError here but later removed it following @mroeschke's suggestion https://github.com/pandas-dev/pandas/pull/26400#discussion_r287456062. `assert False` was added later to prevent mypy error.